### PR TITLE
[WFLY-8613] Create test case for Elytron Aggregate Realm

### PR DESCRIPTION
**Pull request by olukas**
https://issues.jboss.org/browse/WFLY-8613
https://issues.jboss.org/browse/JBEAP-10466

### CODE REVIEW

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/AggregateRealmTestCase.java [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+        public void tearDown(ManagementClient mc, String string) throws Exception {
+            try (CLIWrapper cli = new CLIWrapper(true)) {
+                removeAggregateRealmAndRelatedResources(mc, cli, AGGREGATE_REALM_DIFFERENT_TYPE_NAME);
+                removeAggregateRealmAndRelatedResources(mc, cli, AGGREGATE_REALM_SAME_TYPE_NAME);
+                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s:remove()", FILESYSTEM_REALM_AUTHN_NAME));
+                cli.sendLine(String.format("/subsystem=elytron/properties-realm=%s:remove()", PROPERTIES_REALM_AUTHZ_NAME));
+                cli.sendLine(String.format("/subsystem=elytron/properties-realm=%s:remove()", PROPERTIES_REALM_AUTHN_NAME));
+            } finally {
+                removePropertiesFiles();
+            }
```
Comment (kwart) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/AggregateRealmTestCase.java]: Maybe `ServerReload.reloadIfRequired()` would be safer here.

**Review by kwart — CHANGES_REQUESTED** [review]

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/AggregateRealmTestCase.java [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+                    "/subsystem=undertow/application-security-domain=%1$s:add(http-authentication-factory=%1$s)",
+                    name));
+        }
+
+        private void removeAggregateRealmAndRelatedResources(ManagementClient mc, CLIWrapper cli, String name) throws Exception {
+            cli.sendLine(String.format("/subsystem=undertow/application-security-domain=%s:remove()", name));
+            cli.sendLine(String.format("/subsystem=elytron/http-authentication-factory=%s:remove()", name));
+            cli.sendLine(String.format("/subsystem=elytron/security-domain=%s:remove()", name));
+            cli.sendLine(String.format("/subsystem=elytron/aggregate-realm=%s:remove()", name));
+            ServerReload.reloadIfRequired(mc.getControllerClient());
```
Comment (kwart) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/AggregateRealmTestCase.java]: Move this call just after the 2nd call of `removeAggregateRealmAndRelatedResources` method in the `tearDown`. IMO, it's not necessary to do it for every single aggregate realm.

**Comment by olukas** [comment]
@kwart Fixed. Thanks

**Review by kwart — APPROVED** [review]

**Review by kabir — APPROVED** [review]
Approving paperwork

**Pull request merged**